inc/util.h

```diff
@@ -114,6 +114,7 @@ void amiga_init(void);
 #define B_RIGHT 2
 #define B_SELECT 4
 #define B_PROCESSED 8
+#define B_AUTODISP 16    /* auto sync or polarity has changed */
```

Comment:
I think you can just use B_PROCESSED to trigger a run through the config code. You don't need to add a new flag.

Reply:
Cool - I’ll check it out

Reply:
In fact, you shouldn't poke `buttons` at all, as it races `button_time_fn()` ISR. Instead you should have a `bool_t` in the main loop that gets set when auto-sync changes setup. And then the `if (buttons)` at the end of the main loop should become `if (buttons || your_new_bool)`.

Reply:
Agreed - was just unpicking the existing logic in the last hour
Will do

Reply:
Further thought: `button_time_fn()` will set B_PROCESSED every 5ms anyway. And that will cause `config_process()` to be called. Do you really need to trigger that call in a more timely fashion?

ie. it seems to me that all the B_AUTODISP related chunks could simply be discarded from your patch?

Reply:
The challenge is to repaint the screen and uart menu only when required.
I’ve gone the global flag in main and the checks for it in config.c